e2xgrader/server_extensions/formgrader/handlers.py

```diff
@@ -1,10 +1,106 @@
 import os
-
 from tornado import web
 from nbgrader.api import MissingEntry
+from nbgrader.apps.api import NbGraderAPI
 from nbgrader.server_extensions.formgrader.base import BaseHandler, check_xsrf, check_notebook_dir
 from nbgrader.server_extensions.formgrader.handlers import SubmissionNavigationHandler as NbgraderSubmissionNavigationHandler
-from ...exporters import GradeTaskExporter, GradeNotebookExporter, GradeAssignmentExporter
+from ...exporters import GradeTaskExporter, GradeNotebookExporter, GradeAssignmentExporter, GradeSelectedAssignmentExporter
+from e2xgrader.apps.api import E2X_Gradebook
+import json
+from multiprocessing import Process, Value
+
+
+autograde_flag = Value('b', False)
+autograde_progress = Value('d', 0.0)
+
+
+def autograder(assignment_id):
+    autograde_flag.value = True
+    db_url = 'sqlite:///' + os.path.join(os.getcwd(), 'gradebook.db')
+    gb = NbGraderAPI()
+    students = list(gb.get_submitted_students(assignment_id))
+    total_students = len(students)
+    for idx, student in enumerate(students):
+        autograde_command = "nbgrader autograde " + assignment_id + " --student " + student + " --force"
+        # autograde_command = "nbgrader autograde " + assignment_id + " --student " + student + " --cell-id \"test_sum test_diff\" --force"
+        os.system(autograde_command)
+        autograde_progress.value = round((idx + 1) * 100 / total_students)
+    autograde_progress.value = 0.0
+    autograde_flag.value = False
+
+
+class AutogradeAll(BaseHandler):
```

Comment:
This should be moved to apihandlers. Same for all the other handlers here

Reply:
All handlers associated with 'New Autograder' moved to 'apihandlers.py'